pytorch_lightning/plugins/checkpoint/checkpoint.py

```diff
@@ -0,0 +1,24 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Dict, Union
+
+
+class CheckpointIOPlugin(ABC):
+    @abstractmethod
+    def save_checkpoint(self, checkpoint: Dict[str, Any], path: Union[str, Path]) -> None:
```

Comment:
```suggestion
    def save_checkpoint(self, checkpoint: Dict[str, Any], path: Union[str, Path], storage_options: Optional[Mapping] = None) -> None:
```

we could forward `storage_options` as a black-box dict containing per-plugin per-instance specific information for further options to customize saving and loading. @tchaton this could be another way to represent `map_location`